src/common/hooks/use-asset-transfer.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  createAddress,
+  createAssetInfo,
+  FungibleConditionCode,
+  makeContractCall,
+  makeStandardFungiblePostCondition,
+  PostCondition,
+  standardPrincipalCVFromAddress,
+  uintCV,
+} from '@stacks/transactions';
+import BN from 'bn.js';
+import { useRecoilCallback } from 'recoil';
+import { makeFungibleTokenTransferState } from '@store/recoil/transfers';
+
+interface PostConditionsOptions {
+  contractAddress: string;
+  contractName: string;
+  assetName: string;
+  stxAddress: string;
+  amount: number;
+}
+
+function makePostConditions(options?: PostConditionsOptions) {
```

Comment:
Any reason not to throw an error here? When would you want to make a post condition without options?

Reply:
Agreed - also, it seems odd to call this `makePostConditions`, but it only takes options for one PC, and returns an array of one PC. I know we are only making on PC for internal transfers. Still, I think this should just be called `makePostCondition`, and take mandatory options, and returns one PC.